# Code 16: Data Analysis
## Scott Currie - 12/17/2018

## Video Game Sales

In [1]:
import pandas as pd
import numpy as np
import datetime as dt



In [2]:
df = pd.read_csv('./vgsales.csv')

In [3]:
df.head(10)

Rank                       Name Platform    Year         Genre Publisher  \
0     1                 Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2          Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3             Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4          Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5   Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   
5     6                     Tetris       GB  1989.0        Puzzle  Nintendo   
6     7      New Super Mario Bros.       DS  2006.0      Platform  Nintendo   
7     8                   Wii Play      Wii  2006.0          Misc  Nintendo   
8     9  New Super Mario Bros. Wii      Wii  2009.0      Platform  Nintendo   
9    10                  Duck Hunt      NES  1984.0       Shooter  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37  
5     23.20      2.26      4.22         0.58         30.26  
6     11.38      9.23      6.50         2.90         30.01  
7     14.03      9.20      2.93         2.85         29.02  
8     14.59      7.06      4.70         2.26         28.62  
9     26.93      0.63      0.28         0.47         28.31

### 1. Which company is the most common video game publisher?- 
- Get a series of all publishers
- Group by publisher name
- Get the max

In [4]:
publishers = df.Publisher.groupby(df.Publisher)

In [5]:
publishers.head()

0                           Nintendo
1                           Nintendo
2                           Nintendo
3                           Nintendo
4                           Nintendo
15            Microsoft Game Studios
16              Take-Two Interactive
17              Take-Two Interactive
23              Take-Two Interactive
24              Take-Two Interactive
28       Sony Computer Entertainment
29                        Activision
31                        Activision
33                        Activision
34                        Activision
35                        Activision
38              Take-Two Interactive
43            Microsoft Game Studios
47       Sony Computer Entertainment
52       Sony Computer Entertainment
54       Sony Computer Entertainment
60                           Ubisoft
62            Microsoft Game Studios
65            Microsoft Game Studios
66       Sony Computer Entertainment
68                           Ubisoft
72            Microsoft Game Studios
7

In [18]:
publishers.count().sort_values(ascending=False)

Publisher
Electronic Arts                           1351
Activision                                 975
Namco Bandai Games                         932
Ubisoft                                    921
Konami Digital Entertainment               832
THQ                                        715
Nintendo                                   703
Sony Computer Entertainment                683
Sega                                       639
Take-Two Interactive                       413
Capcom                                     381
Atari                                      363
Tecmo Koei                                 338
Square Enix                                233
Warner Bros. Interactive Entertainment     232
Disney Interactive Studios                 218
Unknown                                    203
Eidos Interactive                          198
Midway Games                               198
505 Games                                  192
Microsoft Game Studios                     189
D3P

### Electronic Arts appears 1351 times.

### 2. What is the most common platform?
- Get series of platforms
- Group by platform
- Sort descending

In [21]:
platforms = df.Platform.groupby(df.Platform)

In [23]:
platforms.count().sort_values(ascending=False)

Platform
DS      2163
PS2     2161
PS3     1329
Wii     1325
X360    1265
PSP     1213
PS      1196
PC       960
XB       824
GBA      822
GC       556
3DS      509
PSV      413
PS4      336
N64      319
SNES     239
XOne     213
SAT      173
WiiU     143
2600     133
NES       98
GB        98
DC        52
GEN       27
NG        12
SCD        6
WS         6
3DO        3
TG16       2
PCFX       1
GG         1
Name: Platform, dtype: int64

### Nintendo DS is the most common platform with 2163 titles.

### 3. What is the most common genre?
- Get a series of genres
- Group by genre
- Sort descending

In [24]:
genres = df.Genre.groupby(df.Genre)

In [73]:
most_common_genre = df['Genre'].value_counts().idxmax()

In [74]:
print(f'Most common genre: { most_common_genre }')

Most common genre: Action


### Action genre is most common with 3316 titles.

### 4. What are the top 20 highest grossing games?

In [75]:
df.sort_values(by=['Global_Sales'], ascending=False)[:20]

Rank                                          Name Platform    Year  \
0      1                                    Wii Sports      Wii  2006.0   
1      2                             Super Mario Bros.      NES  1985.0   
2      3                                Mario Kart Wii      Wii  2008.0   
3      4                             Wii Sports Resort      Wii  2009.0   
4      5                      Pokemon Red/Pokemon Blue       GB  1996.0   
5      6                                        Tetris       GB  1989.0   
6      7                         New Super Mario Bros.       DS  2006.0   
7      8                                      Wii Play      Wii  2006.0   
8      9                     New Super Mario Bros. Wii      Wii  2009.0   
9     10                                     Duck Hunt      NES  1984.0   
10    11                                    Nintendogs       DS  2005.0   
11    12                                 Mario Kart DS       DS  2005.0   
12    13                   Pokemon Gold/Pokemon Silver       GB  1999.0   
13    14                                       Wii Fit      Wii  2007.0   
14    15                                  Wii Fit Plus      Wii  2009.0   
15    16                            Kinect Adventures!     X360  2010.0   
16    17                            Grand Theft Auto V      PS3  2013.0   
17    18                 Grand Theft Auto: San Andreas      PS2  2004.0   
18    19                             Super Mario World     SNES  1990.0   
19    20  Brain Age: Train Your Brain in Minutes a Day       DS  2005.0   

           Genre               Publisher  NA_Sales  EU_Sales  JP_Sales  \
0         Sports                Nintendo     41.49     29.02      3.77   
1       Platform                Nintendo     29.08      3.58      6.81   
2         Racing                Nintendo     15.85     12.88      3.79   
3         Sports                Nintendo     15.75     11.01      3.28   
4   Role-Playing                Nintendo     11.27      8.89     10.22   
5         Puzzle                Nintendo     23.20      2.26      4.22   
6       Platform                Nintendo     11.38      9.23      6.50   
7           Misc                Nintendo     14.03      9.20      2.93   
8       Platform                Nintendo     14.59      7.06      4.70   
9        Shooter                Nintendo     26.93      0.63      0.28   
10    Simulation                Nintendo      9.07     11.00      1.93   
11        Racing                Nintendo      9.81      7.57      4.13   
12  Role-Playing                Nintendo      9.00      6.18      7.20   
13        Sports                Nintendo      8.94      8.03      3.60   
14        Sports                Nintendo      9.09      8.59      2.53   
15          Misc  Microsoft Game Studios     14.97      4.94      0.24   
16        Action    Take-Two Interactive      7.01      9.27      0.97   
17        Action    Take-Two Interactive      9.43      0.40      0.41   
18      Platform                Nintendo     12.78      3.75      3.54   
19          Misc                Nintendo      4.75      9.26      4.16   

    Other_Sales  Global_Sales  
0          8.46         82.74  
1          0.77         40.24  
2          3.31         35.82  
3          2.96         33.00  
4          1.00         31.37  
5          0.58         30.26  
6          2.90         30.01  
7          2.85         29.02  
8          2.26         28.62  
9          0.47         28.31  
10         2.75         24.76  
11         1.92         23.42  
12         0.71         23.10  
13         2.15         22.72  
14         1.79         22.00  
15         1.67         21.82  
16         4.14         21.40  
17        10.57         20.81  
18         0.55         20.61  
19         2.05         20.22

### 5. For North American video game sales, what’s the median?

In [29]:
na_sales = df.NA_Sales.median()

In [30]:
na_sales

0.08

In [31]:
print(f'Median game sales in North America are { na_sales } millions.')

Median game sales in North America are 0.08 millions.


### Provide a secondary output showing ‘about’ ten games surrounding the median sales output

In [35]:
median_sellers = df[df['NA_Sales'] == 0.08]

In [38]:
median_sellers[:10]

Rank                                               Name Platform  \
446    447                                  Dragon Warrior IV      NES   
497    498        World Soccer Winning Eleven 7 International      PS2   
1617  1619                             Farming Simulator 2015       PC   
1926  1928                          Pro Evolution Soccer 2008     X360   
2067  2069  Winning Eleven: Pro Evolution Soccer 2007 (All...     X360   
2373  2375                           Phantasy Star Portable 2      PSP   
2579  2581                               The Sims 2: Castaway      PSP   
3186  3188                                     SingStar Queen      PS2   
3503  3505                                         Top Spin 3      PS3   
3703  3705               Sonic & All-Stars Racing Transformed      PS3   

        Year         Genre                     Publisher  NA_Sales  EU_Sales  \
446   1990.0  Role-Playing              Enix Corporation      0.08      0.00   
497   2003.0        Sports  Konami Digital Entertainment      0.08      1.24   
1617  2014.0    Simulation        Focus Home Interactive      0.08      1.02   
1926  2007.0        Sports  Konami Digital Entertainment      0.08      0.90   
2067  2006.0        Sports  Konami Digital Entertainment      0.08      0.90   
2373  2009.0  Role-Playing                          Sega      0.08      0.11   
2579  2007.0    Simulation               Electronic Arts      0.08      0.46   
3186  2009.0          Misc   Sony Computer Entertainment      0.08      0.12   
3503  2008.0        Action          Take-Two Interactive      0.08      0.37   
3703  2012.0        Racing                          Sega      0.08      0.33   

      JP_Sales  Other_Sales  Global_Sales  
446       3.03         0.01          3.12  
497       1.13         0.45          2.90  
1617      0.00         0.13          1.23  
1926      0.04         0.05          1.07  
2067      0.02         0.00          1.00  
2373      0.62         0.06          0.88  
2579      0.00         0.25          0.80  
3186      0.00         0.44          0.63  
3503      0.00         0.12          0.57  
3703      0.01         0.11          0.54

### 6. For the top-selling game of all time, how many standard deviations above/below the mean are its sales for North America?

In [51]:
top_na_seller = df.sort_values(by=['NA_Sales'], ascending=False).NA_Sales[0]

In [52]:
top_na_seller

41.49

In [55]:
na_sales_mean = df.NA_Sales.mean()

In [56]:
na_sales_mean

0.26466742981082064

In [57]:
na_sales_std = df.NA_Sales.std()

In [87]:
na_sales_std

0.8166830292988796

In [59]:
print(f'Wii Sports is { (top_na_seller - na_sales_mean) / na_sales_std } standard deviations above the mean.')

Wii Sports is 50.47898767479108 standard deviations above the mean.


### 7. The Nintendo Wii seems to have outdone itself with games. How does its average number of sales compare with all of the other platforms?

In [92]:
platforms = df.groupby(df.Platform)
platforms['Global_Sales'].mean().sort_values(ascending=False)

Platform
GB      2.606633
NES     2.561939
GEN     1.050370
SNES    0.837029
PS4     0.827679
X360    0.774672
2600    0.729925
PS3     0.720722
Wii     0.699404
N64     0.686144
XOne    0.662254
PS      0.610920
PS2     0.581046
WiiU    0.572448
3DS     0.486169
GBA     0.387470
DS      0.380254
GC      0.358561
XB      0.313422
SCD     0.311667
DC      0.307115
PC      0.269604
PSP     0.244254
WS      0.236667
SAT     0.194162
PSV     0.149952
NG      0.120000
TG16    0.080000
GG      0.040000
3DO     0.033333
PCFX    0.030000
Name: Global_Sales, dtype: float64